onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -52,7 +53,195 @@ static void BuildFusedKernelDef(KernelDefBuilder& builder, const IndexedSubGraph
+/// </summary>
+/// <param name="graph"></param>
+/// <param name="modified"></param>
+/// <param name="current_ep"></param>
+/// <param name="logger"></param>
+/// <returns></returns>
+static Status TransformLayout(Graph& graph, bool& modified,
+                              IExecutionProvider& current_ep) {
+  // sub graph recurse will be added later
+  auto api_graph = MakeApiGraph(graph, current_ep.GetAllocator(0, OrtMemTypeDefault), nullptr);
+  std::unordered_set<std::string_view> layout_sensitive_ops = onnx_layout_transformation::GetLayoutSensitiveOps();
+
+  for (std::unique_ptr<onnx_layout_transformation::api::NodeRef>& node : api_graph->Nodes()) {
+    if (layout_sensitive_ops.count(node->OpType())) {
+      if (node->GetExecutionProviderType() != current_ep.Type()) {
+        continue;
+      }
+
+      auto domain = node->Domain();
+      // Skip if domain is incorrect
+      if (domain != kOnnxDomain && domain != kOnnxDomainAlias && domain != kMSDomain) {
+        continue;
+      }
+
+      // if already transformed then change the domain to kMSNHWCDomain this way the EP
+      // knows this op is in the expected format.
+      if (node->GetAttributeIntDefault("channels_last", 0) == 1) {
+        onnx_layout_transformation::SwapNodeOpTypeAndDomain(*api_graph, *node, node->OpType(), kMSNHWCDomain);
+        // Changing the domain for the node requires creating a new node and replacing the old one
+        // therefore set the modified flag.
+        modified = true;
+        continue;
+      }
+
+      // Skip if unknown rank
+      auto shape = NodeFromApiNode(*node).InputDefs()[0]->Shape();
+      if (shape == nullptr) {
+        continue;
+      }
+
+      // Convert to channels last
+      size_t rank = shape->dim_size();
+
+      bool has_channel_last_attr = node->GetAttributeInt("channels_last") != nullopt ? true : false;
+      if (has_channel_last_attr) {
+        node->SetAttributeInt("channels_last", 1);
+      }
+
+      auto input_perm = onnx_layout_transformation::ChannelFirstToLastPerm(rank);
+      auto output_perm = onnx_layout_transformation::ChannelLastToFirstPerm(rank);
+
+      // Except for resize and convolution ops, all the other layout sensitive ops only require layout transformation 
+      // for 0th input and output. For resize, add the other relevant inputs which need conversion. For Conv - layout 
+      // transformer only converts layout for 0th input, weights should be handled by every EP.
+      if (node->OpType() == "Resize") {
+        // Older versions of resize have a bug where ROI and Scales cannot be made empty inputs. To handle this case, 
+        // we need to jump a few extra hoops to make sure its inputs are correctly handled. Current code skips 
+        // layout conversion for ROI becasue it needs special handling as ROI size is 2*rank. 
+        // Enable passing in ROI for layout conversion when an EP which supports ROI starts using layout transformer.
```

Comment:
nit: because

[https://marketplace.visualstudio.com/items?itemName=EWoodruff.VisualStudioSpellCheckerVS2022andLater](https://marketplace.visualstudio.com/items?itemName=EWoodruff.VisualStudioSpellCheckerVS2022andLater) is very helpful if you use VS.